src/client/boundaries.rs

```diff
@@ -677,27 +838,29 @@ impl Boundaries {
             boundary_characters: HashMap::new(),
         }
     }
-    pub fn add_rect(&mut self, rect: &dyn Pane) {
+    pub fn add_rect(&mut self, rect: &dyn Pane, should_be_colored: bool, input_mode: InputMode) {
         if rect.x() > 0 {
             let boundary_x_coords = rect.x() - 1;
             let first_row_coordinates = self.rect_right_boundary_row_start(rect);
             let last_row_coordinates = self.rect_right_boundary_row_end(rect);
             for row in first_row_coordinates..last_row_coordinates {
                 let coordinates = Coordinates::new(boundary_x_coords, row);
                 let mut symbol_to_add = if row == first_row_coordinates && row != 0 {
-                    BoundarySymbol::new(boundary_type::TOP_LEFT)
+                    BoundarySymbol::new(boundary_type::TOP_LEFT, should_be_colored)
```

Comment:
If there's already a `colored_borders` method on the `Pane` trait, why not use it here instead of passing `should_be_colored`?

Reply:
oops, seems like I forgot to remove colored borders that I added initially, wasn't really needed for me for this implementation. Do you think I should go with it instead?

Reply:
My intuition says it will be cleaner, but maybe I'm wrong :)